src/Razor/src/Microsoft.AspNetCore.Razor.Language/TagHelperMatchingConventions.cs

```diff
@@ -138,6 +140,18 @@ public static bool SatisfiesBoundAttributeIndexer(string name, BoundAttributeDes
                 name.StartsWith(descriptor.IndexerNamePrefix, StringComparison.OrdinalIgnoreCase);
         }
 
+        public static bool SatisfiesBoundAttributeParameter(string name, BoundAttributeDescriptor parent, BoundAttributeParameterDescriptor descriptor)
+        {
+            if (name != null && name.IndexOf(':') != -1)
+            {
+                var segments = name.Split(':');
+                return (SatisfiesBoundAttributeName(segments[0], parent) || SatisfiesBoundAttributeIndexer(segments[0], parent)) &&
```

Comment:
Break this piece into multiple if -> return statements to ease debugability. Back in TagHelper days I found myself diving through this code all too often and it being debuggable was highly valuable.